### DISCUSSION TITLE
slide sorter view -  menu / overview of all slides

### DISCUSSION THREAD

**Discussion by ivanov** [marp-team/marp#544] [Ideas]
Thank you for making marp!

It would be nice to have a way of getting an overview of all of the slides, so speakers could quickly navigate across the whole presentation (Particularly useful for question and answer section at the end, where going back to particular slide comes in handy, be it in the presentation given, or in the extra pack of slides that the speaker tacked on at the end of the presentation). 

I previously used such functionality in `deck.js` - you can [see a live demo by going to this page, clicking inside the presentation container, and hitting the 'm' key](http://imakewebthings.com/deck.js/introduction/)

#42 alludes to an approach to this kind of functionality, though I don't know if that's been implemented, as I use `marp-cli`.

The [inline SVG slide porition of the documentation](https://marpit.marp.app/inline-svg) suggests that such a feature was anticipated, but it's not clear to me how to apply it. 

I hacked something up with some custom CSS and a bit of javascript to at least get the view, to convince myself that this shouldn't be too complicated.
![image](https://github.com/user-attachments/assets/25bb8136-0888-45ce-8ea4-a4cad66c66c1)

In my hack, using `@marp-team/marp-cli v4.0.0 (w/ @marp-team/marp-core v4.0.0)` , which for me does inline svg out of the box, I tried removing `bespoke-marp-inactive` class from all slides, adding the `-active` class, but didn't succeed in getting the result I wanted until I removed the `bespoke-marp-slide` class so I could get my custom `section` tag CSS to apply. I struggled probably because I need a more specific CSS,. Another hacky approach might be to ~~ab~~use the `?view=next#2`  query parameters and do a bunch of iframe embeds. 

There's certainly got to be a better way,  but I thought I'd write this up to get some guidance, because right now I should be working on my presentation instead of messing with the software :)

**Comment by yhatt** [comment]
Similar to:

- #326 
- marp-team/marp-cli#464
